### DISCUSSION TITLE
Show igate in circuit diagram

### DISCUSSION THREAD

**Discussion by ty-zhao** [QuantumBFS/Yao.jl#497] [Q&A] [answered]
Hi all,

The identity gates `igate` are hidden in the plot made with `YaoPlots.vizcircuit()` by default. Is there a way to show it as a letter 'I' in a square box?

**Comment by GiggleLiu** [comment, marked as answer]
Yes, just wrap it with the LabelBlock.

https://github.com/QuantumBFS/Yao.jl/blob/c23f5154af5cb387d314bbab27ac21c470cd798e/lib/YaoPlots/src/helperblock.jl#L9

Alternatively, you can pirate the following interface (not recommended)
https://github.com/QuantumBFS/Yao.jl/blob/c23f5154af5cb387d314bbab27ac21c470cd798e/lib/YaoPlots/src/vizcircuit.jl#L291

**Reply by ty-zhao** [reply]
The `addlabel()` function in [helperblock.jl](https://github.com/QuantumBFS/Yao.jl/blob/c23f5154af5cb387d314bbab27ac21c470cd798e/lib/YaoPlots/src/helperblock.jl#L9) would do the job. Thanks!
